server/sonar-server/src/main/java/org/sonar/server/measure/custom/ws/CreateAction.java

```diff
@@ -0,0 +1,291 @@
+        "The project id or the project key must be provided. The metric id or the metric key must be provided.<br/>" +
+        "Requires 'Administer System' permission or 'Administer' permission on the project.")
+      .setSince("5.2")
+      .setPost(true)
+      .setHandler(this);
+
+    action.createParam(PARAM_PROJECT_ID)
+      .setDescription("Project id")
+      .setExampleValue("ce4c03d6-430f-40a9-b777-ad877c00aa4d");
+
+    action.createParam(PARAM_PROJECT_KEY)
+      .setDescription("Project key")
+      .setExampleValue("org.apache.hbas:hbase");
+
+    action.createParam(PARAM_METRIC_ID)
+      .setDescription("Metric id")
+      .setExampleValue("16");
+
+    action.createParam(PARAM_METRIC_KEY)
+      .setDescription("Metric key")
+      .setExampleValue("ncloc");
+
+    action.createParam(PARAM_VALUE)
+      .setRequired(true)
+      .setDescription(measureValueDescription())
+      .setExampleValue("47");
+
+    action.createParam(PARAM_DESCRIPTION)
+      .setDescription("Description")
+      .setExampleValue("Team size growing.");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    DbSession dbSession = dbClient.openSession(false);
+    String description = request.param(PARAM_DESCRIPTION);
+    long now = system.now();
+
+    try {
+      ComponentDto component = searchProject(dbSession, request);
+      MetricDto metric = searchMetric(dbSession, request);
+      checkPermissions(component);
+      checkMeasureDoesNotExistAlready(dbSession, component, metric);
+      CustomMeasureDto measure = new CustomMeasureDto()
+        .setComponentUuid(component.uuid())
+        .setComponentId(component.getId())
+        .setMetricId(metric.getId())
+        .setDescription(description)
+        .setCreatedAt(now);
+      setMeasureValue(measure, request, metric);
+      dbClient.customMeasureDao().insert(dbSession, measure);
+      dbSession.commit();
+
+      JsonWriter json = response.newJsonWriter();
+      writeMeasure(json, measure, component, metric, request.mandatoryParam(PARAM_VALUE));
+      json.close();
+    } finally {
+      MyBatis.closeQuietly(dbSession);
+    }
+  }
+
+  private void checkPermissions(ComponentDto component) {
+    if (userSession.hasGlobalPermission(GlobalPermissions.SYSTEM_ADMIN)) {
+      return;
+    }
+
+    userSession.checkLoggedIn().checkProjectUuidPermission(UserRole.ADMIN, component.projectUuid());
+  }
+
+  private void checkMeasureDoesNotExistAlready(DbSession dbSession, ComponentDto component, MetricDto metric) {
+    int nbMeasuresOnSameMetricAndMeasure = dbClient.customMeasureDao().countByComponentIdAndMetricId(dbSession, component.uuid(), metric.getId());
+    if (nbMeasuresOnSameMetricAndMeasure > 0) {
+      throw new ServerException(HttpURLConnection.HTTP_CONFLICT, String.format("A measure already exists for project id '%s' and metric id '%d'", component.uuid(), metric.getId()));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 181 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)